### ISSUE TITLE
[Authentication Docs & TypeScript] Suggestions for minor improvements regarding signup example

### ISSUE THREAD

**Issue by Philzen** [redwoodjs/redwood#5036]
While working through Chapter 4 of the tutorial i came across https://redwoodjs.com/docs/authentication#signuphandler

One minor problem i had was being a bit confused at first about [the explanation](https://github.com/redwoodjs/redwood/blob/main/docs/docs/authentication.md?plain=1#L161) and the following code snippet, which does not return a string at all – until i realized that this refers to signup page code, while the other snippets above it refer to `auth.js`. It would be helpful to add the docusaurus ` ``` title="api/src/functions/auth.ts"` annotation to the code blocks so it's easier to understand which snippets belong where. (Suggestion No 1)

---

Trying out to return a string in `auth.ts` worked as expected at first, however a typescript warning appears towards the end of the file:

```terminal
Type '{ handler: ({ username, hashedPassword, salt, userAttributes }: { username: any; hashedPassword: any; salt: any; userAttributes: any; }) => string; errors: { fieldMissing: string; usernameTaken: string; }; }' is not assignable to type '{ handler: (signupHandlerOptions: SignupHandlerOptions) => Promise<any>; errors?: { fieldMissing?: string; usernameTaken?: string; }; }'.
  The types returned by 'handler(...)' are incompatible between these types.
    Type 'string' is not assignable to type 'Promise<any>'.ts(2322)
[DbAuthHandler.d.ts(90, 5): ]()The expected type comes from property 'signup' which is declared here on type 'DbAuthHandlerOptions'
```
TypeScript actually excelled here b/c it challenged me to think for a second and realize that the `DbAuthHandlerOptions` interface was actually fine, but it was me doing something really stupid :facepalm: – of course one can need return the Promise, but append `.then(() => "Some Feedback")` to return a string in order to keep the new user logged out.

However, the code would still compile just returning, including the warning and work just as fine (errors are still triggered and shown in the frontend). Should we amend the interface of `DbAuthHandlerOptions` to include `string` return type, or at maybe just add an example to the docs using `then()`? (suggestion No 2)

**Comment by Philzen** [comment]
The same problem arises at the end of chapter 4: https://redwoodjs.com/docs/tutorial/chapter4/deployment#the-signup-problem where learners are advised to return `false` from the api call:

```bash
Type '{ handler: ({ username, hashedPassword, salt, userAttributes }: { username: any; hashedPassword: any; salt: any; userAttributes: any; }) => boolean; errors: { fieldMissing: string; usernameTaken: string; }; }' is not assignable to type '{ handler: (signupHandlerOptions: SignupHandlerOptions) => Promise<any>; errors?: { fieldMissing?: string; usernameTaken?: string; }; }'.
  The types returned by 'handler(...)' are incompatible between these types.
    Type 'boolean' is not assignable to type 'Promise<any>'.ts(2322)
[DbAuthHandler.d.ts(90, 5): ]()The expected type comes from property 'signup' which is declared here on type 'DbAuthHandlerOptions'
```

Actually, advising to return a string such as "Signup disabled to all you hacking snollygosters!" would be much more enjoyable :sunglasses: : 

(and also a good learning exercise in-line with Authentication docs, which don't say anything about returning a boolean)

**Comment by cannikin** [comment]
Good catch! I'm not a TS person so I've never noticed this message. Want to take a stab at a PR that updates the types you're talking about so they don't warn?
